src/app.js

```diff
@@ -465,6 +468,23 @@ App.prototype.constructI18n = function( translations ) {
 	return new Jed( translations );
 };
 
+/**
+ * Registers a custom data callback.
+ *
+ * @param {Function} callback The callback to register.
+ *
+ * @returns {void}
+ */
+App.prototype.registerCustomDataCallback = function( callback ) {
+	if( ! this.callbacks.custom ) {
+		this.callbacks.custom = [];
+	}
+
+	if ( isFunction( callback ) ) {
+		this.callbacks.push( callback );
```

Comment:
Shouldn't this be `this.callbacks.custom.push( callback )`? Or what's the purpose of 
```
	if( ! this.callbacks.custom ) {
		this.callbacks.custom = [];
	}
```
?